lib/pengine/variant.h

```diff
@@ -85,7 +85,18 @@ typedef struct
 
 } container_port_t;
 
+enum container_type {
+        CONTAINER_TYPE_UNKNOWN,
+        CONTAINER_TYPE_DOCKER,
+        CONTAINER_TYPE_RKT
+};
+
+#define CONTAINER_TYPE_UNKNOWN_S "unknown"
+#define CONTAINER_TYPE_DOCKER_S  "docker"
+#define CONTAINER_TYPE_RKT_S     "rkt"
+
 typedef struct container_variant_data_s {
+        enum container_type type;
```

Comment:
New fields have to be added to the end of a struct to retain ABI backward compatibility. With Pacemaker 2.0 we're going to break ABI compatibility, so we can move it back to the top then, but for now (1.1.18) it should be at the end.